### ISSUE TITLE
Dust.js has started outputting 'line,2 line,4 line,13 line,6' in weird places

### ISSUE THREAD

**Issue by sup3rman** [linkedin/dustjs#556]
Come DustJS 2.6.0 it has started outputting line,[number] in random places. It seems to be affected by the `dust.config.whitespace` setting. Weirdly enough I got the unwanted output when `dust.config.whitespace = false` on a DO droplet (Node v0.10.33), while I got it when `dust.config.whitespace = true` on my workstation. Downgrading to v2.5.1 resolved the problem.

I precompile all my templates.

I am pressed for time the coming weeks, but if nobody else is experiencing this I could dig some more to find out exactly where the error occurs.

**Comment by sethkinast** [comment]
Thanks for the report. Can you provide a template that shows the issue?

**Comment by sup3rman** [comment]
Hi, seems to be for empty blocks. Not that you should have these all over the place, but many of my boilerplate logic seems to have them anyway.

```
dust.compile('{<test}\n{/test}{<test2}\n{/test2}', 'test')
```
outputs
```
(function(dust){dust.register("test",body_0);var blocks={"test":body_1,"test2":body_2};function body_0(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk;}body_0.__dustBody=!0;function body_1(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk.w("\nline,1");}body_1.__dustBody=!0;function body_2(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk.w("\nline,2");}body_2.__dustBody=!0;return body_0;})(dust);
```

**Comment by sethkinast** [comment]
Thanks, I know the commit to look at. Looks like this is an edge case that has never been tested.

That should be enough info for the problem to be fixed. Thanks again for the report.

**Comment by sethkinast** [comment]
Repro:

```
echo "{<test}\n{/test}{<test2}\n{/test2}" | node_modules/.bin/dustc --name test --whitespace
INFO: No files to compile. Write template code and use ^D to exit
(function(dust){dust.register("test",body_0);var blocks={"test":body_1,"test2":body_2};function body_0(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk.w("\nline,1");}body_0.__dustBody=!0;function body_1(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk.w("\\n");}body_1.__dustBody=!0;function body_2(chk,ctx){ctx=ctx.shiftBlocks(blocks);return chk.w("\\n");}body_2.__dustBody=!0;return body_0;})(dust);
```

**Comment by sethkinast** [comment]
Target commit https://github.com/linkedin/dustjs/commit/374bbdb6c377307d67adea9377ad4c47f82984e5

**Comment by sethkinast** [comment]
Incidentally the `dust.config.whitespace` setting only affects the template compilation, not the rendering-- so that's why you were seeing it happen on both your desktop and server, because I assume you sent precompiled templates to the server.

**Comment by sup3rman** [comment]
Nah I wasnt, my Node applications compile all templates on start up. But
good to hear you know what's up.

**Comment by sethkinast** [comment]
We have released 2.6.1 with a fix. Thanks for taking the time to report!

**Comment by sup3rman** [comment]
Cheers!
